### ISSUE TITLE
[fin v1.41.0] Xdebug triggers on each page request (even when no breakpoints defined)

### ISSUE THREAD

**Issue by ApacheEx** [docksal/docksal#420]
Hi,
after upgrading to the newer version (v1.41.0) I've noticed that xdebug doesn't work as it was before. I think the issue can be related to these changes 9d24e39bb10a95184cc54995456904aca633668b (but not sure)

I'm using:
- PHPStorm 2017.3
- MacOS 10.13.2
- Docksal via VirtualBox (boot2docker)
- `DOCKSAL_STACK=default`

currently, it's quite hard to debug something. Any ideas?

**Comment by achekulaev** [comment]
So what is exactly the issue? Xdebug always triggers (triggers meaning works, not stops) per every page request. That's why it's best to disable it if you don't use it.

**Comment by ApacheEx** [comment]
@achekulaev 

somehow Xdebug can't stop on a breakpoint. I always see
![image](https://user-images.githubusercontent.com/1918108/33825102-30fb2e42-de69-11e7-8f02-2b0ea630384f.png)


My PHPStorm settings:
![image](https://user-images.githubusercontent.com/1918108/33825137-4fca9f7e-de69-11e7-8e1d-bd5b237aa76f.png)

**Comment by achekulaev** [comment]
@ApacheEx please post `fin config`

**Comment by ApacheEx** [comment]
<details> 
  <summary>Output for `fin config` </summary>
```
---------------------
COMPOSE_PROJECT_NAME_SAFE: yawave
COMPOSE_FILE:
/Users/apacheex/.docksal/stacks/volumes-bind.yml
/Users/apacheex/.docksal/stacks/stack-default.yml
/Users/apacheex/WEB/yawave/.docksal/docksal.yml
ENV_FILE:
/Users/apacheex/WEB/yawave/.docksal/docksal.env
/Users/apacheex/WEB/yawave/.docksal/docksal-local.env

PROJECT_ROOT: /Users/apacheex/WEB/yawave
DOCROOT: docroot
VIRTUAL_HOST: yawave.docksal
VIRTUAL_HOST_ALIASES: *.yawave.docksal
IP: 192.168.64.100
MYSQL: 192.168.64.100:33061

Docker Compose configuration
---------------------
services:
  adminer:
    image: adminer
    labels:
      io.docksal.virtual-host: pma.yawave.docksal
    ports:
    - 8080:8080/tcp
  cli:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      DOCROOT: docroot
      HOST_GID: '20'
      HOST_UID: '501'
      PHP_IDE_CONFIG: serverName=yawave.docksal
      XDEBUG_CONFIG: remote_connect_back=0 remote_host=192.168.64.1
      XDEBUG_ENABLED: '1'
    hostname: cli
    image: docksal/cli:1.3-php5
    volumes:
    - docksal_ssh_agent:/.ssh-agent:ro
    - /home/docker
    - project_root:/var/www:rw,nocopy
    - /Users/apacheex/WEB/yawave-app/public:/var/www/docroot/yawave-app/public:rw
    - /Users/apacheex/WEB/yawave-sdk/build1:/var/www/docroot/yawave-sdk/build1:rw
  db:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      MYSQL_DATABASE: default
      MYSQL_PASSWORD: user
      MYSQL_ROOT_PASSWORD: root
      MYSQL_USER: user
    hostname: db
    image: docksal/db:1.1-mysql-5.5
    ports:
    - 33061:3306/tcp
    volumes:
    - project_root:/var/www:ro,nocopy
  memcached:
    command:
    - -m
    - '128'
    dns:
    - 192.168.64.100
    - 8.8.8.8
    hostname: memcached
    image: memcached:1.4-alpine
  web:
    depends_on:
      cli:
        condition: service_started
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      APACHE_BASIC_AUTH_PASS: null
      APACHE_BASIC_AUTH_USER: null
      APACHE_DOCUMENTROOT: /var/www/docroot
    hostname: web
    image: docksal/web:2.1-apache2.4
    labels:
      io.docksal.project-root: /Users/apacheex/WEB/yawave
      io.docksal.virtual-host: yawave.docksal,*.yawave.docksal
    volumes:
    - project_root:/var/www:ro,nocopy
    - /Users/apacheex/WEB/yawave-app/public:/var/www/docroot/yawave-app/public:rw
    - /Users/apacheex/WEB/yawave-sdk/build1:/var/www/docroot/yawave-sdk/build1:rw
version: '2.1'
volumes:
  docksal_ssh_agent:
    external: true
    name: docksal_ssh_agent
  project_root:
    driver: local
    driver_opts:
      device: /Users/apacheex/WEB/yawave
      o: bind
      type: none

---------------------
```
</details>

**Comment by achekulaev** [comment]
Should work fine. @lmakarov any ideas?

**Comment by ApacheEx** [comment]
After adding a `PHP Remote Debug` like this:
![image](https://user-images.githubusercontent.com/1918108/33882132-d1551b92-df3f-11e7-9dc2-5a28fd9e0c4e.png)

And clicking on a green icon (fyi, `Start Listening for PHP Debug Connections` should be off):
![image](https://user-images.githubusercontent.com/1918108/33882193-fe4fbe0e-df3f-11e7-9e1b-0b5fa564a974.png)


It works again (without any warnings):
![image](https://user-images.githubusercontent.com/1918108/33882315-5cf655c6-df40-11e7-9c02-0d7c17c53cf1.png)


but does it really a what we need as solution?

**Comment by lmakarov** [comment]
@ApacheEx I remember this issue now. 

The "Incoming Connection From Xdebug" dialogue in PHPStorm (http://take.ms/Ez7k4) won't show up when `PHP_IDE_CONFIG: serverName=...` is set. That variable is necessary for console scripts debugging, where serverName is not available (since there is no web server involved), so PHPStorm does not know which server debug configuration to use. 

I guess, we should drop it from the default stack and if anyone needs to debug console scripts, then will have to manually add it there. This way we make the experience straightforward for the majority of users (who are doing web debugging).

Alternatively add instructions to configured the debug server settings manually, but that would be a downgrade in user experience.

**Comment by ApacheEx** [comment]
I just overriden `PHP_IDE_CONFIG` like this (so it's empty)
![image](https://user-images.githubusercontent.com/1918108/33906710-6451839a-df8b-11e7-802d-0a21ab1e6d36.png)


after that: all works fine again 🎉  even without Xdebug helper.

I think it's better to remove this option by default.

**Comment by lmakarov** [comment]
Fixed and rolled out in [v1.6.1](https://github.com/docksal/docksal/releases/tag/v1.6.1)
